support/testsupport/src/main/java/com/hotels/styx/support/LatchesAndBarriers.java

```diff
@@ -0,0 +1,57 @@
+/*
+  Copyright (C) 2013-2018 Expedia Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.hotels.styx.support;
+
+import java.util.concurrent.BrokenBarrierException;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.CyclicBarrier;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static java.lang.String.format;
+import static java.lang.Thread.currentThread;
+
+/**
+ * Makes `CountDownLatch` more convenient for unit tests.
+ */
+public final class LatchesAndBarriers {
+    private LatchesAndBarriers() {
+    }
+
+    public static void await(CountDownLatch latch, long timeout, TimeUnit unit) throws IllegalStateException {
```

Comment:
To be honest, I don't understand why we think we need to capture InterruptedException here at all. The test can fail due to an InterruptedException and, in that case, it makes all the sense to have it in the signature of the Test itself.
If we still wanted to wrap the code of await() , we could have a method that receives a lambda instead of having to duplicate the code for CountdownLatch and CyclicBarrier.

Reply:
I think the problem is with the lambda expressions. Without this utility you would be forced to try/catch inside lambda expressions.

Reply:
As Mikko says, the problem is that `InterruptedException` is a checked exception, and the interfaces our lambdas are implementing do not allow checked exceptions to be thrown.

Reply:
Oh, sure. I think  some of the code inside the tests uses the Latch outside a lambda. 

 Then my suggestion of using a method that receives a lambda (with InterruptedException in the signature) and propagates InterruptedException as a RuntimeException (but keeping the Thread.interrupt() ) makes more sense  than wrapping specific classes.

Reply:
Hi @dvlato. It no longer wraps a class. It is just a static helper method that accepts a `CountDownLatch` as an argument. IMO it should just set the interrupted flag and propagate an InterruptedException wrapped inside a RuntimeException.